connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -432,7 +439,9 @@ public void onCompletion(Throwable error, ConsumerRecord<String, byte[]> record)
 
             final SchemaAndValue value;
             try {
-                value = converter.toConnectData(topic, record.value());
+                synchronized (lock) {
```

Comment:
I'm ok with not synchronizing access to converters. I wasn't sure whether these were the only two implementations we use. Jackson and  String serialization are thread-safe. I found that Avro might be used in an unsafe way, but we don't use it like that.